internal/chat/domain/chat_service.go

```diff
@@ -173,7 +174,38 @@ func (s *ChatServiceImpl) CreateChat(ctx context.Context, chat Chat) (*Chat, err
 }
 
 func (s *ChatServiceImpl) UpdateChat(ctx context.Context, chat Chat) (*Chat, error) {
-	updatedChat, err := s.chatRepo.UpdateChat(ctx, chat)
+	var (
+		setClause []string
+		values    []any
+	)
+	user := ctx.Value("user").(*domain.User)
+
+	existingChat, err := s.GetChat(ctx, chat.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if existingChat == nil {
+		return nil, errors.NewNotFoundError(common.ChatDomain)
+	}
+
+	if existingChat.CreatedBy != user.ID {
+		return nil, errors.NewForbiddenError()
+	}
+
+	if chat.Name != "" {
+		values = append(values, chat.Name)
+		setClause = append(setClause, fmt.Sprintf("name = $%d", len(values)))
```

Comment:
```suggestion
		setClause = append(setClause, fmt.Sprintf("name = $%d", len(values)))
```

This approach isn’t ideal. The service layer shouldn’t be aware of repository logic. What happens if we switch databases? I suggest hardcoding the logic in the repository and only calling it when changes occur. Alternatively, we could update it each time using the previous version.